tiledb/sm/cpp_api/group_experimental.h

```diff
@@ -61,12 +61,35 @@ class Group {
       const Context& ctx,
       const std::string& group_uri,
       tiledb_query_type_t query_type)
-      : ctx_(ctx) {
-    tiledb_ctx_t* c_ctx = ctx.ptr().get();
-    tiledb_group_t* group;
-    ctx.handle_error(tiledb_group_alloc(c_ctx, group_uri.c_str(), &group));
-    group_ = std::shared_ptr<tiledb_group_t>(group, deleter_);
-    ctx.handle_error(tiledb_group_open(c_ctx, group, query_type));
+      : Group(ctx, group_uri, query_type, nullptr) {
+  }
+
+  /**
+   * @brief Constructor. Sets a config to the group and opens it for the given
+   * query type. The destructor calls the `close()` method.
+   *
+   * **Example:**
+   *
+   * @code{.cpp}
+   * // Open the group for reading
+   * tiledb::Context ctx;
+   * tiledb::Config cfg;
+   * cfg["rest.username"] = "user";
+   * cfg["rest.password"] = "pass";
+   * tiledb::Group group(ctx, "s3://bucket-name/group-name", TILEDB_READ, cfg);
+   * @endcode
+   *
+   * @param ctx TileDB context.
+   * @param group_uri The group URI.
+   * @param query_type Query type to open the group for.
+   * @param config COnfiguration parameters
+   */
+  Group(
+      const Context& ctx,
+      const std::string& group_uri,
+      tiledb_query_type_t query_type,
+      const Config& config)
```

Comment:
Can I just add a `const std::optional<Config>& config = nullopt` to the existing constructor? I'm not sure if it would break binary compatibility.

Reply:
That seems be ok in practice (though not guaranteed by spec): https://stackoverflow.com/a/41764100/508431

Reply:
I'll double-check with conda test builds.

Reply:
~~I _think_ it's OK; the C API is one with the stable ABI, we don't expect a library to export the C++ API and another library to copy it, right?~~

Hmm, `std::optional` does not work with references, let's keep it as-is and have two constructors.